pkg/visibility/api/install.go

```diff
@@ -20,42 +22,32 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apiserver/pkg/registry/rest"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 
-	"sigs.k8s.io/kueue/apis/visibility/v1alpha1"
+	visibilityv1alpha1 "sigs.k8s.io/kueue/apis/visibility/v1alpha1"
+	visibilityv1beta1 "sigs.k8s.io/kueue/apis/visibility/v1beta1"
 	"sigs.k8s.io/kueue/pkg/queue"
-	apirest "sigs.k8s.io/kueue/pkg/visibility/api/rest"
+	restv1alpha1 "sigs.k8s.io/kueue/pkg/visibility/api/rest/v1alpha1"
+	restv1beta1 "sigs.k8s.io/kueue/pkg/visibility/api/rest/v1beta1"
 )
 
 var (
-	Scheme = runtime.NewScheme()
-
-	Codecs = serializer.NewCodecFactory(Scheme)
-
+	Scheme         = runtime.NewScheme()
+	Codecs         = serializer.NewCodecFactory(Scheme)
 	ParameterCodec = runtime.NewParameterCodec(Scheme)
 )
 
 func init() {
-	utilruntime.Must(v1alpha1.AddToScheme(Scheme))
-	utilruntime.Must(Scheme.SetVersionPriority(v1alpha1.GroupVersion))
+	utilruntime.Must(visibilityv1alpha1.AddToScheme(Scheme))
+	utilruntime.Must(visibilityv1beta1.AddToScheme(Scheme))
+	utilruntime.Must(Scheme.SetVersionPriority(visibilityv1alpha1.GroupVersion, visibilityv1beta1.GroupVersion))
 	metav1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
 }
 
-// Install installs API scheme defined in apis/v1alpha1 and registers storage
+// Install installs API scheme and registers storages
 func Install(server *genericapiserver.GenericAPIServer, kueueMgr *queue.Manager) error {
-	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(v1alpha1.GroupVersion.Group, Scheme, ParameterCodec, Codecs)
-	pendingWorkloadsInCqREST := apirest.NewPendingWorkloadsInCqREST(kueueMgr)
-	cqREST := apirest.NewCqREST()
-	pendingWorkloadsInLqREST := apirest.NewPendingWorkloadsInLqREST(kueueMgr)
-	lqREST := apirest.NewLqREST()
-
-	visibilityServerResources := map[string]rest.Storage{
```

Comment:
What happens to this?

Reply:
The same situation here. We need to keep both `v1alpha1` and `v1beta1` and to optimize code I moved it to separate files.

Reply:
https://github.com/kubernetes-sigs/kueue/blob/b765d06caeae4c380f7f9555092e960ac6e31450/pkg/visibility/api/rest/v1alpha1/storage.go#L25-L32

Reply:
I see, sgtm, we may just call it `NewStorage`.